maven-core/src/main/java/org/apache/maven/lifecycle/internal/LifecycleDependencyResolver.java

```diff
@@ -240,8 +240,11 @@ private Set<Artifact> getDependencies(
         scopesToCollect = new HashSet<>(scopesToCollect);
         scopesToCollect.addAll(scopesToResolve);
 
-        DependencyFilter collectionFilter = new ScopeDependencyFilter(null, negate(scopesToCollect));
-        DependencyFilter resolutionFilter = new ScopeDependencyFilter(null, negate(scopesToResolve));
+        Collection<String> collectScopesToExclude = negate(scopesToCollect);
+        Collection<String> resolveScopesToExclude = negate(scopesToResolve);
+
+        DependencyFilter collectionFilter = new ScopeDependencyFilter(null, collectScopesToExclude);
+        DependencyFilter resolutionFilter = new ScopeDependencyFilter(null, resolveScopesToExclude);
```

Comment:
can we just use the scopes to collect / resolve instead of the ones to exclude and give them as a negative test in the filter ?
i.e.:
```
DependencyFilter collectionFilter = new ScopeDependencyFilter(scopesToCollect);
DependencyFilter resolutionFilter = new ScopeDependencyFilter(scopesToResolve);
```